src/adk/schema/applications/application.json

```diff
@@ -1,71 +1,79 @@
 {
-  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$schema": "http://json-schema.org/draft-07/schema#",
   "type": "array",
-  "items": [
-    {
-      "type": "object",
-      "properties": {
-        "title": {
-          "type": "string"
-        },
-        "description": {
-          "type": "string"
-        },
-        "values": {
-          "type": "array",
-          "items": [
-            {
-              "type": "object",
-              "properties": {
-                "name": {
-                  "type": "string"
-                },
-                "default_value": {
-                  "type": "integer"
-                },
-                "minimum_value": {
-                  "type": "integer"
-                },
-                "maximum_value": {
-                  "type": "integer"
-                },
-                "unit": {
-                  "type": "string"
-                },
-                "scale_value": {
-                  "type": ["number", "string"]
-                }
-              },
-              "required": [
-                "name",
-                "default_value",
-                "minimum_value",
-                "maximum_value",
-                "unit",
-                "scale_value"
+  "items": {
+    "type": "object",
+    "properties": {
+      "title": {
+        "type": "string"
+      },
+      "description": {
+        "type": "string"
+      },
+      "values": {
+        "type": "array",
+        "items": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string"
+            },
+            "default_value": {
+              "type": [
+                "string",
+                "number"
               ]
-            }
-          ]
-        },
-        "input_type": {
-          "type": "string"
-        },
-        "roles": {
-          "type": "array",
-          "items": [
-            {
+            },
+            "minimum_value": {
+              "type": [
+                "string",
+                "number"
+              ]
+            },
+            "maximum_value": {
+              "type": [
+                "string",
+                "number"
+              ]
+            },
+            "unit": {
               "type": "string"
+            },
+            "scale_value": {
+              "type": [
+                "string",
+                "number"
+              ]
             }
+          },
+          "required": [
+            "name",
+            "default_value",
+            "minimum_value",
+            "maximum_value",
+            "unit",
+            "scale_value"
           ]
         }
       },
-      "required": [
-        "title",
-        "description",
-        "values",
-        "input_type",
-        "roles"
-      ]
-    }
-  ]
+      "input_type": {
+        "type": "string"
+      },
+      "roles": {
+        "type": "array",
+        "patternProperties": {
+          "[A-Z][-a-zA-Z0-9_]*": {
```

Comment:
I think this change might make it impossible to run some of our own example applications with the ADK?

The applications: Blind Quantum Computing, Magic Square and CHSH all have roles that do not start with a capital leter.

Reply:
Ouch. I looked at the schemas defined in api-router and copied what was there.
I looked up the workings of patternProperties and we use it totally wrong:
patternProperties says when a property matches the regular expression, the value must match the schema defined for this patternProperties. So when role was an object and had a property "Role-1" it ts value should be a string. I will change this back and add a ticket for the schemas in api-router.